docs/advanced_usage.rst

```diff
@@ -34,6 +34,32 @@ hooks, as these can be run on any platform. However, if you intend for your
 template to only be run on a single platform, a shell script (or `.bat` file
 on Windows) can be a quicker alternative.
 
+.. note::
+    Make sure your hook scripts work in a robust manner. If a hook script fails
+    (that is, `if it finishes with a nonzero exit status
+    <https://docs.python.org/2/library/sys.html#sys.exit>`_), the project
+    generation will stop and the generated directory will be cleaned up.
+
+Example: Validating template variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is an example of script that validates a template variable
+before generating the project, to be used as ``hooks/pre_gen_project.py``::
```

Comment:
I feel sorry for being this picky... :speak_no_evil: 

[PEP8 Imports](https://www.python.org/dev/peps/pep-0008/#imports) recommends the following:

```python
import re
import sys
```

Also you could enable syntax highlighting in rst via:

```
to be used as ``hooks/pre_gen_project.py``:

.. code-block:: python

    import re
```